File: 205. Isomorphic Strings.java
public class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> hm = new HashMap<Character, Character>();
        
        for (int i = 0; i < s.length(); i++) {
            char a = s.charAt(i);
            char b = t.charAt(i);
            
            if (hm.containsKey(a)) {
                if (hm.get(a) == b) {
                    
                } else {
                    return false;
                }
            } else {
                if (hm.containsValue(b)) {
                    return false;
                } else {
                    hm.put(a,b);
                }
            }
        }
        
        return true;
    }
}

File: 205. Isomorphic Strings.py
class Solution:
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        mapping = {}
        
        if len(s) != len(t):
            return False
        
        for i in range(len(s)):
            if s[i] in mapping:
                if t[i] != mapping[s[i]]:
                    return False
            elif t[i] in mapping.values():
                return False
            else:
                mapping[s[i]] = t[i]
        
        return True
